pg_yregress/instance.c

```diff
@@ -204,14 +204,25 @@ yinstance_connect_result yinstance_connect(yinstance *instance) {
   return yinstance_connect_failure;
 }
 
-void yinstance_start(yinstance *instance) {
+void yinstance_prepare(yinstance *instance) {
   if (instance->managed) {
+    char datadir_template[] = "pg_yregress_XXXXXX";
+    char *datadir = mkdtemp(datadir_template);
+    if (datadir == NULL) {
+      fprintf(stderr, "Failed to create temporary directory: %s\n", strerror(errno));
+      return;
+    }
+    char *heap_datadir = strdup(datadir);
+    instance->info.managed.datadir = (iovec_t){.base = heap_datadir, .len = strlen(heap_datadir)};
+  }
+}
 
-    char datadir[] = "pg_yregress_XXXXXX";
-    mkdtemp(datadir);
```

Comment:
previously `mkdtemp`'s return value was not used which seems like a bug. `mkdtemp` can fail, in which case the return value would be `NULL`